source/agora/node/Ledger.d

```diff
@@ -144,13 +134,14 @@ public class Ledger
 
     ***************************************************************************/
 
-    private inout(Transaction)* findTransaction (Hash tx_hash) inout @safe
+    private Output* findOutput (Hash tx_hash, size_t index) @safe
     {
         foreach (ref block; this.ledger)
         {
             if (block.header.tx_hash == tx_hash)
             {
-                return &block.tx;
+                if (index < block.tx.outputs.length)
+                    return &block.tx.outputs[index];
```

Comment:
What happen if I send a transaction which reference a previous tx but with `index` == `size_t.max` ?

Reply:
Hmm what did you mean by this? That we should limit the index? the `if` check here should do the job for now, no?

Reply:
I mean that you'll keep iterating the ledger, even though there's no way you get another match.